File: src/my_py_pkg/my_py_pkg/hardware_status_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from my_robot_interfaces.msg import HardwareStatus

class HardwareStatusPublisherNode(Node):
    def __init__(self):
        super().__init__("hardware_status_publisher")
        self.hw_status_pub_ = self.create_publisher(HardwareStatus, "hardware_status", 10)
        timer_period = 1.0  # seconds
        self.timer = self.create_timer(timer_period, self.publish_hw_status)
        self.get_logger().info("Hardware Status publisher has been started.")

    def publish_hw_status(self):
        msg = HardwareStatus()
        msg.are_motors_ready =  True  
        msg.temperature = 45.0  
        msg.debug_message = "All systems operational."
        self.hw_status_pub_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = HardwareStatusPublisherNode()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/hardware_status_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "my_robot_interfaces/msg/hardware_status.hpp"

using namespace std::chrono_literals;

class HardwareStatusPublisher : public rclcpp::Node
{
public:
    HardwareStatusPublisher() : Node("hardware_status_publisher")
    {
        hw_status_pub_ = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>("hardware_status", 10);
        timer_ = this->create_wall_timer(
            1000ms, 
            std::bind(&HardwareStatusPublisher::publish_hw_status, this));
        RCLCPP_INFO(this->get_logger(), "Hardware Status Publisher has been started.");
    }

private:

    void publish_hw_status()
    {
        auto msg = my_robot_interfaces::msg::HardwareStatus();
        msg.are_motors_ready = true;
        msg.temperature = 45.0;
        msg.debug_message = "All systems operational.";
        hw_status_pub_->publish(msg);
    }

    rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr hw_status_pub_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<HardwareStatusPublisher>());
    rclcpp::shutdown();
    return 0;
}